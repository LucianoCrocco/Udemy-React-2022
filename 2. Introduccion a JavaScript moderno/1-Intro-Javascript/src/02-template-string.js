//Template Strings
const nombre = 'Luciano';
const apellido = 'Crocco';

console.log (`${nombre} ${apellido}
${getSaludo(nombre)}`);

function getSaludo(nombre){
    return 'Hola ' + nombre;
}