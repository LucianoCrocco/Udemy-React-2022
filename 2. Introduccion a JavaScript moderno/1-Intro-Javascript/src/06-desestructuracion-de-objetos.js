// 7. DesestructuraciÃ³n de objetos
