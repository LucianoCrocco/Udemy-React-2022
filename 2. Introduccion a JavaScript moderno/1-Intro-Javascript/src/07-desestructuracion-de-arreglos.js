// 7. DesestructuraciÃ³n de arreglos

const personajes = ['Goku', 'Vegeta', 'Trunks'];

/*
console.log(personajes[0]);
console.log(personajes[1]);
console.log(personajes[2]);
*/
const [ , , p3]  = personajes

console.log(p3);

const retoronaArreglo = () => {
    return ['ABC', 123];
}

const [letras, numeros] = retoronaArreglo();
console.log(letras, numeros);

//Tarea
// 1. El primer valor del arreglo se llamara nombre
// 2. Se llamara a setNombre
const UserState = (valor) => {
    return [valor, () => console.log('Hola Mundo')];
} 

const arr = UserState('Goku');
console.log(arr);
arr[1]();

const [ nombre, setNombre] = UserState(p3);
console.log(nombre);
setNombre();