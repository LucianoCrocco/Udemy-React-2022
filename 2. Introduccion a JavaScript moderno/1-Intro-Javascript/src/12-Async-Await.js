const getImage = async () => {
    try {
        const api_key = "Gqmh1Vt5CDRSIfZxED6UZApWI2ozHjtd";
        const resp = await fetch(`https://api.giphy.com/v1/gifs/random?api_key=${api_key}`);
        const { data } = await resp.json();
        const { url } = data.images.original;    
        const img = document.createElement('img');
        img.src = url;
        document.body.append(img);
    } catch (error) {
        console.log(error);
    }

}

getImage();