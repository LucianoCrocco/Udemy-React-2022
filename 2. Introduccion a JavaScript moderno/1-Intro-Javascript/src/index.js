const api_key = "Gqmh1Vt5CDRSIfZxED6UZApWI2ozHjtd";

const peticion = fetch(`https://api.giphy.com/v1/gifs/random?api_key=${api_key}`);

peticion
.then(res => res.json())
.then(({data}) =>{
    const { url } = data.images.original;
    const img = document.createElement('img');
    img.src = url;
    document.body.append(img);
})
.catch(console.warn);