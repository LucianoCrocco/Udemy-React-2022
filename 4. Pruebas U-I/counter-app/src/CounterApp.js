import React, {useState} from "react";
import { Fragment } from "react";
import PropTypes from 'prop-types';


const CounterApp = ( {value} ) => {

    const [counter, setCounter] = useState(value);

    //Al re renderizar me permite volver a generar el componente y cambiar el valor de la constante.

    const handleAdd = () => {
        //setCounter(counter + 1);
        setCounter((c) => c+1);
    };

    const handleSubtract = () => {
        setCounter(counter - 1);
    }

    const handleReset = () => {
        setCounter(value);
    }

    return (
        <Fragment>
            <h1>CounterApp</h1>
            <h2>{counter}</h2>

            <button onClick={handleAdd}>+1</button>
            <button onClick={handleReset}>Reset</button>
            <button onClick={handleSubtract}>-1</button>
        </Fragment>
    );
}


CounterApp.propTypes = {
    value : PropTypes.number.isRequired
}

CounterApp.defaultProps = {
    value : 1
}


export default CounterApp;