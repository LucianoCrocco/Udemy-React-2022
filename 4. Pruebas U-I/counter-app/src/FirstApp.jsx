import { Fragment } from "react";
import PropTypes from "prop-types";

export const FirstApp = ( {title, subTitle, name} ) => {
    return (<Fragment>
    
        <h1>{title}</h1>
        <p>{subTitle}</p>
        <p>{name}</p>
   
    </Fragment>);
}

FirstApp.propTypes = {
    title : PropTypes.string.isRequired,
    subTitle : PropTypes.string.isRequired
}

FirstApp.defaultProps = {
    name : "Luciano Crocco",
    subTitle : "No hay subtitulo",
    title : "No hay titulo"
}