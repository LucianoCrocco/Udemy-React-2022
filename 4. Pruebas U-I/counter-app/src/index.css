html, body {
    background-color: #21232a;
    color: white;
    font-family: Helvetica, Arial, sans-serif;
    font-size: 1.3em;
    padding: 70px;
}

button{
    border: 2px solid red;
    border-radius: 10px;
    font-size: 1.3em;
    margin-right: 10px;
    padding: 10px 30px;
}