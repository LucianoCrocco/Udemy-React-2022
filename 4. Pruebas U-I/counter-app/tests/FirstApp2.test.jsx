import {render, screen} from '@testing-library/react';
import {FirstApp} from '../src/FirstApp';

describe('Pruebas en <FirstApp />', () => { 
    const titulo = "No hay titulo";
    const subtitulo = "No hay subtitulo"
    test("Debe hacer match con el snapshot", () => {
        const {container} = render( <FirstApp /> );
        expect(container).toMatchSnapshot();
    })

    test("Debe mostrar el mensaje 'No hay titulo'", () => {
        render( <FirstApp /> );
        expect(screen.getByText(titulo)).toBeTruthy();
    })

    test("Debe mostrar el titulo en un h1", () => {
        render( <FirstApp /> );
        expect(screen.getByRole("heading", {level : 1}).innerHTML).toContain(titulo);
    })

    test("Debe mostrar el subtitulo 'No hay subtitulo'", () => {
        render( <FirstApp /> );
        expect(screen.getByText(subtitulo).innerHTML).toContain(subtitulo);
    })

})