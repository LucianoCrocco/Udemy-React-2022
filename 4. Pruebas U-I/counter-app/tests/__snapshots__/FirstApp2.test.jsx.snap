// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp /> Debe hacer match con el snapshot 1`] = `
<div>
  <h1>
    No hay titulo
  </h1>
  <p>
    No hay subtitulo
  </p>
  <p>
    Luciano Crocco
  </p>
</div>
`;
